app/Http/Traits/DataAccessApplicationHelpers.php

```diff
@@ -418,4 +418,19 @@ public function groupApplicationsByProject(LengthAwarePaginator $applications):
             $page
         );
     }
+
+    public function returnApplicationsInProject(LengthAwarePaginator $applications): LengthAwarePaginator
+    {
+        $applications->getCollection()->transform(function ($application) {
+            $sameProject = DataAccessApplication::where('project_id', $application['project_id'])
+                ->get();
+            $teams = array();
+            foreach ($sameProject as $app) {
+                $teams = array_merge($teams, $app['teams']->toArray());
+            }
+            $application['teams'] = $teams;
```

Comment:
Also to avoid defining arrays, merging, using toArray, you can use something like:
```
   $teams = $sameProject->flatMap(function ($app) {
            return $app->teams;
   })->unique('id')->values(); 
   
   $application->teams = $teams;

```

^^ something like that is marginally more readable IMO

Reply:
Neat, I suppose that is also more memory efficient, given that it defines fewer temporary arrays?